testutils/src/main/java/com/microsoft/identity/internal/testutils/kusto/EstsKustoUtils.java

```diff
@@ -0,0 +1,193 @@
+     * @param testResultFileName the file containing android test results
+     */
+    public static void ingestAndroidClientTestResults(@NonNull final String testResultFileName) {
+        final IngestionMapping ingestionMapping = new IngestionMapping(
+                ESTS_KUSTO_CLIENT_TEST_TABLE_INGESTION_MAPPING_ANDROID,
+                IngestionMapping.IngestionMappingKind.Csv
+        );
+
+        ingest(
+                EstsKustoClientTestTableData.ESTS_KUSTO_CLIENT_RESULT_TABLE_NAME,
+                ingestionMapping,
+                testResultFileName
+        );
+    }
+
+    /**
+     * Ingest data contained within the supplied file into the specified table in the ESTS Kusto
+     * Cluster.
+     *
+     * @param tableName        the table into which to ingest
+     * @param ingestionMapping the mapping to use while ingestion
+     * @param fileToIngest     the file that contains the data to ingest
+     */
+    public static void ingest(@NonNull final String tableName,
+                              @NonNull final IngestionMapping ingestionMapping,
+                              @NonNull final String fileToIngest) {
+        final ConnectionStringBuilder connectionStringBuilder = createKustoConnectionString(
+                KustoOperation.Ingest
+        );
+
+        IngestClient client;
+
+        // Creating the client:
+        try {
+            client = IngestClientFactory.createClient(connectionStringBuilder);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Creating the ingestion properties:
+        IngestionProperties ingestionProperties = new IngestionProperties(
+                ESTS_DATABASE_NAME,
+                tableName
+        );
+
+        ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.Table);
+        ingestionProperties.setDataFormat(IngestionProperties.DATA_FORMAT.csv);
+        ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FailuresAndSuccesses);
+        ingestionProperties.setFlushImmediately(true);
+
+        ingestionProperties.setIngestionMapping(ingestionMapping);
+
+        try {
+            FileSourceInfo fileSourceInfo = new FileSourceInfo(fileToIngest, 0);
+            IngestionResult ingestionResult = client.ingestFromFile(fileSourceInfo, ingestionProperties);
+
+            List<IngestionStatus> statuses = ingestionResult.getIngestionStatusCollection();
+
+            int timeoutInSec = 60;
+
+            while (statuses.get(0).status == OperationStatus.Pending && timeoutInSec > 0) {
+                Thread.sleep(1000);
+                timeoutInSec -= 1;
+                statuses = ingestionResult.getIngestionStatusCollection();
+                Log.i(TAG, "Ingestion status: " + statuses.get(0).status.toString());
+            }
+
+            for (IngestionStatus status : statuses) {
+                Log.i(TAG, "Final ingestion status: " + status.status.toString());
```

Comment:
I think we might want to do something else here for failures, otherwise we might forget to look for some number of months.  Not that that's ever happened to any automated system ever...

Reply:
Addressed by throwing a runtime exception on failure